### ISSUE TITLE
Need to fix Fortran "use mpi" module array declarations in APIs such as MPI_WAITALL

### ISSUE THREAD

**Issue by jsquyres** [open-mpi/ompi#9484]
It was recently noticed that some of the array declarations in the ignore TKR `mpi` module have specifically-sized arrays, but the MPI standard says that the arrays should be `(*)`.  The easiest API to note is `MPI_WAITALL`.

All the other Fortran APIs with array dummy parameters should also be checked; I don't know how pervasive the problem is.  This is probably not nearly as daunting a job as it sounds -- just search `ompi/mpi/fortran/use-mpi-ignore-tkr/*.h` for dummy parameters that are arrays that are not `(*)` and double-check them against the MPI standard.

The TKR `mpi` module should also be checked.

------

History:
* This conversation started in #9367
* The other issues identified in that original PR were split off into other PRs to fix, but this issue remains to be fixed.

**Comment by markalle** [comment]
In this PR I just looked at the various "dimension ... array_of_..." lines:
https://github.com/open-mpi/ompi/pull/9566

**Comment by markalle** [comment]
(is in master, and I just rebased the v5.0.x cherry-pick):
https://github.com/open-mpi/ompi/pull/9608

**Comment by jsquyres** [comment]
We fixed this on master and v5.0.x.

Although there are no ABI issues associated with this change, and it's _probably_ low risk to take these changes back to v4.0.x and v4.1.x, we made the executive decision to not take them back to these old branches just because there doesn't seem to be much benefit (no one has complained about this for years).

So we're leaving this note for our future selves: if we need to, we _can_ back-port this to the v4.x branches.  But unless we have to, let's not.
